loadbalancer/algorithm.go

```diff
@@ -41,24 +42,81 @@ var (
 )
 
 func newRoundRobin(endpoints []string) routing.LBAlgorithm {
-	i := time.Now().UnixNano()
-	rand.Seed(i)
+	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
 	return &roundRobin{
-		index: rand.Intn(len(endpoints)),
+		index: rnd.Intn(len(endpoints)),
+		rnd:   rnd,
 	}
 }
 
 type roundRobin struct {
 	mx    sync.Mutex
 	index int
+	rnd   *rand.Rand
+}
+
+func fadeIn(now time.Time, duration time.Duration, ease float64, detected time.Time) float64 {
+	rel := now.Sub(detected)
+	if rel >= duration {
+		return 1
+	}
+
+	return math.Pow(float64(rel)/float64(duration), ease)
```

Comment:
Here I like that `ease` may be <1, not sure if that is useful though:
![x1/8, x1/4, x1/2, x1, x2, x4, x8](https://upload.wikimedia.org/wikipedia/commons/thumb/6/61/Mplwp_roots_01.svg/450px-Mplwp_roots_01.svg.png)

Reply:
I'm not entirely sure about the usefulness either, but there was no reason to prevent it.